l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -212,7 +218,10 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
     }
 
     /// @dev Used to set the assedAddress for a given assetId.
-    function setAssetHandlerAddressInitial(bytes32 _additionalData, address _assetHandlerAddress) external {
+    function setAssetHandlerAddressInitial(
+        bytes32 _additionalData,
+        address _assetHandlerAddress
+    ) external onlyOwnerOrNTV {
         address sender = msg.sender == address(nativeTokenVault) ? NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS : msg.sender;
```

Comment:
sender can only be wither ntv or owner, meaning that assetid can only have owner or ntv as a handler, which I don't think is an expected behavior (as I understand we would want for owner to be able to register tokens with different asset handlers)